File: Java/primoGrado.java
public class primoGrado extends equazione{

    public primoGrado(polinomio sx, polinomio dx) throws erroreGrado{
        super(sx,dx);
        polinomio p= unifica();
        p.ordina();
        razionale r= new razionale(1);
        if(p.getGrado().diverso(r)){
            throw new erroreGrado();
        }
    }

    public String getRisultato(){
        polinomio temp = unifica();
        razionale menoUno = new razionale(-1);
        temp.ordina();
        razionale r = new razionale();
        try{
            r = razionale.moltiplicazioneRazionale(menoUno,razionale.divisioneRazionale(temp.getMonomio_pos(1).getCoefficiente(),temp.getMonomio_pos(0).getCoefficiente()));
        }
        catch(Exception e){}
        return "x = "+r.toString();
    }

    public String getTipo(){
        return "Equazione di primo grado";
    }

    public String getGrado(){
        return Integer.toString(1);
    }

    public primoGrado getDerivata() throws erroreGrado{
        return this;
    }
}

File: C++/Model/primogrado.cpp
#include "primogrado.h"

primoGrado::primoGrado(const polinomio& a,const polinomio& b):
    equazione(a,b){
    if(unifica().getGrado()!=razionale(1))
        throw erroreGrado();
}

string primoGrado::risolvi() const{
    polinomio temp = unifica();
    razionale menoUno(-1,1);
    temp.ordina();
    temp.eliminaZeri();
    razionale r;
    if(temp.size()==1)
        return string("x = 0");
    r = menoUno*temp.getMonomio_pos(1).getCoefficiente()/temp.getMonomio_pos(0).getCoefficiente();
    return string("x = "+r.toString());
}

tipoEquazione primoGrado::getTipo() const{
    return tipoEquazione(0);
}

razionale primoGrado::getGrado() const{
    return razionale(1);
}

primoGrado* primoGrado::derivata() const{
    return 0;
}
